theano/gof/opt.py

```diff
@@ -1131,13 +1131,22 @@ class LocalMetaOptimizer(LocalOptimizer):
 
     """
 
-    def __init__(self, tracks=None, optimizers=()):
-        self._tracks = tracks
+    def __init__(self, optimizers=()):
+        self._tracks = [x for o in optimizers for x in o.tracks()]
         self.optimizers = list(optimizers)
         self.verbose = config.metaopt.verbose
+        self.track_dict = defaultdict(lambda: [])
```

Comment:
Ah you're right, I was confused by the register function and the way the diff is presented.

But yeah it would help reduce duplication to do it that way.